rmgpy/yml.py

```diff
@@ -63,15 +63,37 @@ def write_yml(spcs, rxns, solvent=None, solvent_data=None, path="chem.yml"):
 
 
 def get_mech_dict(spcs, rxns, solvent='solvent', solvent_data=None):
-    result_dict = dict()
-    result_dict["Units"] = dict()
-    result_dict["Phases"] = [dict()]
-    result_dict["Phases"][0]["name"] = "phase"
-    result_dict["Phases"][0]["Species"] = [obj_to_dict(x, spcs) for x in spcs]
-    result_dict["Reactions"] = [obj_to_dict(x, spcs) for x in rxns]
-    if solvent_data:
-        result_dict["Solvents"] = [obj_to_dict(solvent_data, spcs, label=solvent)]
-    return result_dict
+    names = [x.label for x in spcs]
+    for i,name in enumerate(names): #fix duplicate names
+        if names.count(name) > 1:
+            names[i] += "-"+str(names.count(name))
+
+    is_surface = False
+    for spc in spcs:
+        if spc.contains_surface_site():
+            is_surface = True
+            break
+    if not is_surface:
+        result_dict = dict()
+        result_dict["Units"] = dict()
+        result_dict["Phases"] = [dict()]
+        result_dict["Phases"][0]["name"] = "phase"
+        result_dict["Phases"][0]["Species"] = [obj_to_dict(x, spcs, names=names) for x in spcs]
+        result_dict["Reactions"] = [obj_to_dict(x, spcs, names=names) for x in rxns]
+        if solvent_data:
+            result_dict["Solvents"] = [obj_to_dict(solvent_data, spcs, names=names, label=solvent)]
```

Comment:
Currently we can't do solvent corrections for rmg-cat, but this may be something we implement in the future.  In this case, perhaps we add the "Solvents" to `results_dict` regardless of whether `is_surface` is true or not.  If `is_surface` is True, then `solvent_data` should be None, so it's not added anyways, but if we implement solvent corrections for surface mechs in the future, then we should add it to yaml.